#
# From http://rakshasa.no/pipermail/libtorrent-devel/2008-February/001503.html
#
[Libtorrent-devel] fix building with gcc 4.3 and above
Serge Belyshev belyshev at depni.sinp.msu.ru
Fri Feb 29 07:30:00 UTC 2008

    * Previous message: [Libtorrent-devel] Total incoming incorrect
    * Next message: [Libtorrent-devel] fix building rtorrent HEAD without xmlrpc
    * Messages sorted by: [ date ] [ thread ] [ subject ] [ author ]

In gcc 4.3 header dependencies were cleaned up, so libtorrent/rtorrent
do not build anymore. Fixed by the following patches, for libtorrent:

Index: rtorrent-0.8.0/rak/functional.h
===================================================================
--- rtorrent-0.8.0.orig/rak/functional.h
+++ rtorrent-0.8.0/rak/functional.h
@@ -37,6 +37,7 @@
 #ifndef RAK_FUNCTIONAL_H
 #define RAK_FUNCTIONAL_H
 
+#include <stddef.h>
 #include <functional>
 
 namespace rak {
Index: rtorrent-0.8.0/src/command_ui.cc
===================================================================
--- rtorrent-0.8.0.orig/src/command_ui.cc
+++ rtorrent-0.8.0/src/command_ui.cc
@@ -39,6 +39,7 @@
 #include <rak/functional.h>
 #include <rak/functional_fun.h>
 #include <sigc++/adaptors/bind.h>
+#include <ctime>
 
 #include "core/manager.h"
 #include "core/view_manager.h"
Index: rtorrent-0.8.0/src/display/text_element_value.cc
===================================================================
--- rtorrent-0.8.0.orig/src/display/text_element_value.cc
+++ rtorrent-0.8.0/src/display/text_element_value.cc
@@ -38,6 +38,7 @@
 
 #include "globals.h"
 #include "text_element_value.h"
+#include <ctime>
 
 namespace display {
 
Index: rtorrent-0.8.0/src/rpc/exec_file.cc
===================================================================
--- rtorrent-0.8.0.orig/src/rpc/exec_file.cc
+++ rtorrent-0.8.0/src/rpc/exec_file.cc
@@ -37,6 +37,7 @@
 #include "config.h"
 
 #include <string>
+#include <cstring>
 #include <unistd.h>
 #include <rak/path.h>
 #include <sys/types.h>
Index: rtorrent-0.8.0/src/rpc/parse.cc
===================================================================
--- rtorrent-0.8.0.orig/src/rpc/parse.cc
+++ rtorrent-0.8.0/src/rpc/parse.cc
@@ -39,6 +39,7 @@
 #include <locale>
 #include <rak/path.h>
 #include <torrent/exceptions.h>
+#include <cstring>
 
 #include "parse.h"
 

    * Previous message: [Libtorrent-devel] Total incoming incorrect
    * Next message: [Libtorrent-devel] fix building rtorrent HEAD without xmlrpc
    * Messages sorted by: [ date ] [ thread ] [ subject ] [ author ]

More information about the Libtorrent-devel mailing list

